### DISCUSSION TITLE
docs/gateway/

### DISCUSSION THREAD

**Discussion by giscus** [ricsanfre/pi-cluster#258] [Announcements]
# docs/gateway/

How to configure a Raspberry Pi as router/firewall of our Kubernetes Cluster providing connectivity and basic services (DNS, DHCP, NTP, SAN).

https://picluster.ricsanfre.com/docs/gateway/

**Comment by bpartin2009** [comment]
How did you configure your Ansible Control Node, which is on your 192.168.1.0/24 home network, to send the node (10.0.0.1/24) traffic to the Cluster Gateway Pi?
I tried adding a route "sudo route add -net 10.0.0.0/24 gw 192.168.1.95 eth0" but I still can't ping the nodes.

**Comment by bpartin2009** [comment]
I can ping the 10.0.0.1 Cluster Gateway from the 192.168.1.X Ansible Control Node but it seems like the gateway is not forwarding the traffic.

**Reply by ricsanfre** [reply]
Hi @bpartin2009, ICMP traffic to cluster nodes from Ansible Control Node is blocked by firewall rules configured in `gateway`. You can ping the nodes only from gateway node.

**Comment by alber82** [comment]
Hi! I've been trying to deploy your great solution to my cluster for several days. I have been overcoming most of the problems but I cannot launch the "make nodes-setup". I get the following error 
fatal: [node1]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh: connect to host 10.0.0.11 port 22: Connection timed out", "unreachable": true}
In principle, the "make gateway-setup" and everything above worked correctly. I can ping from the gateway and not from the pi-cluster, which according to what I read is normal. But the ansible runner is not able to reach the nodes

**Reply by ricsanfre** [reply]
Hi @alber82 , thanks for your comment.

Did you add the route to lab network 10.0.0.0/24, via your gateway's IP address in your home network, to your local host where docker is running? 
In linux server it should be a command like: 
```shell
route add -net 10.0.0.0/24 gw <gateway's IP address in your home network> <network_dev>

route add -net 10.0.0.0/24 gw 192.168.1.11  eth0
```

In my case, since I am using a Ubuntu Desktop version, I have that permanently configured in my Wifi connection IPv4 settings.

Executing command `ip route` you should see the route

```shell
$ ip route
default via 192.168.1.1 dev wlp2s0 proto dhcp metric 600 
10.0.0.0/24 via 192.168.1.11 dev wlp2s0 proto static metric 1
```


Regards
Ricardo

**Comment by alber82** [comment]
Hello, thanks for the response, it seems that I had a repeated route and it was giving me problems. 
The route line to 10.0.0.0/24 was routed through two different ips.
